relay-metrics/src/aggregator/mod.rs

```diff
@@ -155,50 +164,86 @@ impl Aggregator {
         let mut partitions = HashMap::new();
         let mut stats = HashMap::new();
 
+        let mut re_add = Vec::new();
+
         let now = Instant::now();
+        let ts = UnixTimestamp::from_instant(now.into_std());
+
+        let should_pop = |entry: &QueuedBucket| force || entry.elapsed(now);
+
+        let bucket_interval = self.config.bucket_interval;
+        let cost_tracker = &mut self.cost_tracker;
+
+        let start_bucket_scan = Instant::now();
+        while let Some((key, entry)) = self.buckets.try_pop(|_, entry| should_pop(entry)) {
```

Comment:
nit: I had a hard time reading this loop because of the many "entries", maybe this helps.

```suggestion
        while let Some((key, queued_bucket)) = self.buckets.try_pop(|_, b| should_pop(b)) {
```